syntax = "proto2";

package apollo.drivers.robosense;

import "modules/common/proto/header.proto";

message PointXYZITR {
  optional float x = 1 [default = nan];
  optional float y = 2 [default = nan];
  optional float z = 3 [default = nan];
  optional uint32 intensity = 4 [default = 0];
  optional uint64 timestamp = 5 [default = 0];
  optional uint32 ring = 6 [default = 0];
}

message PointCloud_withFullInfo {
  optional apollo.common.Header header = 1;
  optional string frame_id = 2;
  optional bool is_dense = 3;
  repeated PointXYZITR point = 4;
  optional double measurement_time = 5;
  optional uint32 width = 6;
  optional uint32 height = 7;
}
